### PR TITLE
Medias component

### PR DESCRIPTION
**Pull request by dlaxcess**
Cf issue #182

### CODE REVIEW

File: svelte/components/Global/ContentV2.svelte [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  import Nft from "../Nft/NftV2.svelte";
+  import NftList from "../Nft/NftV2List.svelte";
+
+  export let chainId: number;
+  export let address: string;
+  export let account: string;
+  export let platform: string;
+  export let refreshing: boolean = false;
+  export let refresh: number;
+
```
Comment (zapaz) [inline svelte/components/Global/ContentV2.svelte:13]: Add component description, something like  :

```
  /////////////////////////////////////////////////
  // <ContentV2 {chainId} {address} {account} {platform} {refreshing} {refresh} />
  // Placeholder for ...
  /////////////////////////////////////////////////
```

File: svelte/components/Global/ContentV2.svelte [hunk 2]
```diff
@@ -0,0 +1,41 @@
+<script lang="ts">
+  // import { displayedTokenID } from "main/diplayedNft";
+
+  import Nft from "../Nft/NftV2.svelte";
+  import NftList from "../Nft/NftV2List.svelte";
+
+  export let chainId: number;
+  export let address: string;
+  export let account: string;
```
Comment (zapaz) [inline svelte/components/Global/ContentV2.svelte:9]: account optionnal ? 

same remark for platform and refresh

File: svelte/components/Global/mediasDisplay/MediasDisplayer.svelte [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  import MediaPreview from "./mediasComponents/MediaPreview.svelte";
+  import DisplayTypedMedia from "./mediasComponents/DisplayTypedMedia.svelte";
+
+  /////////////////////////////////////////////////
+  //  <MediasDisplayer {nft} {index} {displayMode}? />
+  // Determin type of media and Display correct component according to the entering parameters of {displayMode}
+  /////////////////////////////////////////////////
+  export let nft: NftType;
+  export let index: number;
+  export let displayMode: { target: string; small: boolean } = { target: "preview", small: true };
```
Comment (zapaz) [inline svelte/components/Global/mediasDisplay/MediasDisplayer.svelte:15]: should not use object as props, better pass 2 props

or pass all options as object : not sure it's a good idea for reactivity

File: svelte/components/Global/mediasDisplay/MediasDisplayer.svelte [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  let alt: string = nft?.name || mediaType;
+
+  $: mediaContentType = nft.contentType?.split("/");
+  $: mediaType = mediaContentType[0];
+  // $: mediaSubtype = mediaContentType[1];
+  $: mediaSrc = nftGetImageLink(nft);
+</script>
+
+{#if "preview" === displayMode.target}
+  <MediaPreview {mediaSrc} {index} {mediaType} {displayMode} {alt} />
```
Comment (zapaz) [inline svelte/components/Global/mediasDisplay/MediasDisplayer.svelte:31]: TS => Type '{ target: string; small: boolean; }' is not assignable to type 'string'

File: svelte/components/Global/mediasDisplay/mediasComponents/DisplayImage.svelte [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+  /////////////////////////////////////////////////
+  //  <DisplayVideo {mediaSrc} {alt}? {height}? />
+  // Display a Image according to its entering parameters
+  /////////////////////////////////////////////////
+  export let mediaSrc: string;
+  export let alt: string = "image";
+  export let height: number = undefined;
+</script>
+
+<img src={mediaSrc} {alt} height={height ? height : undefined} />
```
Comment (zapaz) [inline svelte/components/Global/mediasDisplay/mediasComponents/DisplayImage.svelte:11]: `height={height ? height : undefined}`  =>` {height}`  should work ?

File: svelte/components/Global/mediasDisplay/mediasComponents/DisplayTypedMedia.svelte [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+  import DisplayVideo from "./DisplayVideo.svelte";
+
+  /////////////////////////////////////////////////
+  //  <DisplayTypedMedia {mediaSrc} {index} {mediaType} {displayMode} {alt}? />
+  // Display a media according to its type and entering parameters
+  /////////////////////////////////////////////////
+  export let mediaSrc: string;
+  export let index: number;
+  export let mediaType: string;
+  export let displayMode: { target: string; small: boolean };
```
Comment (zapaz) [inline svelte/components/Global/mediasDisplay/mediasComponents/DisplayTypedMedia.svelte:12]: 2 props instead

File: svelte/components/Global/mediasDisplay/mediasComponents/DisplayTypedMedia.svelte [hunk 7]
```diff
@@ -0,0 +1,27 @@
+<script lang="ts">
+  import DisplayImage from "./DisplayImage.svelte";
+  import DisplayVideo from "./DisplayVideo.svelte";
+
+  /////////////////////////////////////////////////
+  //  <DisplayTypedMedia {mediaSrc} {index} {mediaType} {displayMode} {alt}? />
```
Comment (zapaz) [inline svelte/components/Global/mediasDisplay/mediasComponents/DisplayTypedMedia.svelte:6]: + paused

File: svelte/components/Global/mediasDisplay/mediasComponents/DisplayTypedMedia.svelte [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+  let small: string = displayMode.small ? "small" : "full";
+  let media: string = "list" === displayMode.target ? "media-small" : "media-grid";
+  let gridScale: string = "grid" === displayMode.target ? " a-simul-cursor" : "";
+</script>
+
+<div id="media-{small}-{index}" class="media {media} media-{mediaType}{gridScale}">
+  {#if "image" === mediaType}
+    <DisplayImage {mediaSrc} {alt} />
+  {:else if "video" === mediaType}
+    <DisplayVideo {mediaSrc} small={displayMode.small} options={{ mode: displayMode.target, paused }} />
```
Comment (zapaz) [inline svelte/components/Global/mediasDisplay/mediasComponents/DisplayTypedMedia.svelte:25]: no object as props

File: svelte/components/Nft/NftV2Grid.svelte [hunk 9]
```diff
@@ -0,0 +1,52 @@
+<script lang="ts">
+  import type { NftType } from "lib/ktypes";
+
+  import { nftName, nftOpenSeaUrl, addressSame, getNetwork } from "lib/kconfig";
+
+  import MediasDisplayer from "../Global/mediasDisplay/MediasDisplayer.svelte";
+
+  /////////////////////////////////////////////////
+  //  <NftGrid {nft} {account} {index} {platform}? />
```
Comment (zapaz) [inline svelte/components/Nft/NftV2Grid.svelte:9]: comments and props differs

File: svelte/rollup.config.ts [hunk 10]
```diff
@@ -56,7 +56,14 @@ const toRollupConfig = function (component: string): RollupOptions {
     ],
     plugins: [
       svelte({
-        preprocess: [sveltePreprocess({})],
+        preprocess: [
+          sveltePreprocess({
+            sourceMap: !production,
+            postcss: {
+              plugins: [require("autoprefixer")()]
```
Comment (zapaz) [inline svelte/rollup.config.ts:63]: why use autoprefixer ?

**Review by zapaz — CHANGES_REQUESTED** [review]

File: svelte/rollup.config.ts (hunk 10 above)
Comment (dlaxcess) [reply svelte/rollup.config.ts:63]: It's to permit to use SCSS styles in component, very use full

**Pull request merged**